skimage/color/colorlabel.py

```diff
@@ -145,9 +145,18 @@ def _label2rgb_overlay(label, image=None, colors=None, alpha=0.3,
         The result of blending a cycling colormap (`colors`) for each distinct
         value in `label` with the image, at a certain alpha value.
     """
+    # number of unique labels
+    nlabel = len(np.unique(label))
+
     if colors is None:
-        colors = DEFAULT_COLORS
-    colors = [_rgb_vector(c) for c in colors]
+        if nlabel <= len(DEFAULT_COLORS):
+            colors = DEFAULT_COLORS
+            colors = [_rgb_vector(c) for c in colors]
+        else:
+            colors = np.random.random((nlabel, 3))
```

Comment:
I see some potential issues with this approach:
1. It is not reproducible
2. The contrast between the colors is not guaranteed
3. The method is prone to color collisions.

I would rather suggest to consider adopting (we don't have it as a hrad dependency anymore) one of the qualitative matplotlib colormaps (https://matplotlib.org/3.2.0/tutorials/colors/colormaps.html) and sample the colors from it.

Reply:
(Just to say it's not @mohitpokharna fault, @jni suggested it :) https://github.com/scikit-image/scikit-image/issues/4507#issuecomment-598463715 )

Reply:
@soupault Thank you for pointing out the potential issues. I have added a fix. Please review.

Reply:
@mohitpokharna sorry for not being 100% clear. By adopting I meant literally taking a piece of relevant code for the colormap generation from the matplotlib codebase. As scikit-image we have made a decision to switch matplotlib from a requirement to an optional dependency (https://github.com/scikit-image/scikit-image/issues/2710). Thus, we should avoid importing it in commonly used functions (such as label2rgb). If you have an alternative idea, please, let us know. 👍

Reply:
@soupault How about indicating `colors = [np.asarray(cm.tab20c(x))[:-1] for x in range(n_colors)]` in the documentation? The user could make its own color series if needed...

Reply:
Sorry about all the confusion, everyone! :sweat_smile: Actually the `np.random.random` approach was suggested as an alternative to the user's code, but not something that we would want to do internally. Instead, internally, I think we should use [Glasbey](https://github.com/taketwo/glasbey) + shuffle (+ sampling above some set number of colours, eg 255). But Glasbey is a bit complicated to implement. It might be easier to start from the [paper](https://strathprints.strath.ac.uk/30312/1/colorpaper_2006.pdf) than the code.

Reply:
@mohitpokharna at any rate, thank you very much for attempting this fix!

Reply:
@jni Thank you for pointing out. It makes much more sense now. I will try to see if I can carry it on further.

Reply:
@sciunto not sure if I like that :slightly_smiling_face: . We should rather aim at hiding the complexity.